thanks-bot/bot.js

```diff
@@ -23,13 +23,41 @@ bot.on('ready', function (evt) {
 });
 
 bot.on('message', function (user, userID, channelID, message, evt) {
-    switch(message) {
-        case 'Ассаляму аляйкум':
-            bot.sendMessage({
-                to: channelID,
-                message: 'Уа аляйкум салям',
-            });
-
-            break;
+    const regexp = /\!thanks\b/gmi;
+    const bot_user_id = '523928478341660702';
+    const allowed_channel = '525249065039036426';
+    const authorId = evt.d.author.id;
+    const authorName = evt.d.author.username;
+
+    // Conditions for exiting a function
+    if(channelID !== allowed_channel ||
+        authorId === bot_user_id || !regexp.test(message)){
+        return;
+    }
+
+    // Collecting usernames
+    let mentionedUserNames = evt.d.mentions.map(item => item.username);
+
+    // Collecting user id
+    let mentionedUserId = evt.d.mentions.map(item => item.id);
+
+    // Exit function if no user is mentioned
+    if(mentionedUserId.length === 0){
+        return;
     }
+
+    if (!!mentionedUserId.indexOf(authorId)) {
```

Comment:
`!!` doesn't have any sense. Additionally, use `contains` instead of `indexOf`.
```suggestion
    if (mentionedUserId.indexOf(authorId)) {
```